src/Nox.Generator/Application/Commands/UpsertEnumerationTranslationsCommand.template.cs

```diff
@@ -0,0 +1,109 @@
+
+internal partial class {{upsertCommand}}Handler : {{upsertCommand}}HandlerBase
+{
+	public {{upsertCommand}}Handler(
+        AppDbContext dbContext,
+		NoxSolution noxSolution) : base(dbContext, noxSolution)
+	{
+	}
+}
+internal abstract class {{upsertCommand}}HandlerBase : CommandBase<{{upsertCommand}}, {{enumEntity}}>, IRequestHandler<{{upsertCommand}}, IEnumerable<{{enumAtt.EntityDtoNameForLocalizedEnumeration}}>>
+{
+	public AppDbContext DbContext { get; }
+	private static readonly Nox.Types.CultureCode _defaultCultureCode = Nox.Types.CultureCode.From("{{codeGeneratorState.Solution.Application.Localization.DefaultCulture}}");
+
+	public {{upsertCommand}}HandlerBase(
+        AppDbContext dbContext,
+		NoxSolution noxSolution) : base(noxSolution)
+	{
+		DbContext = dbContext;
+	}
+
+	public virtual async Task<IEnumerable<{{enumAtt.EntityDtoNameForLocalizedEnumeration}}>> Handle({{upsertCommand}} command, CancellationToken cancellationToken)
+	{
+		cancellationToken.ThrowIfCancellationRequested();
+		await OnExecutingAsync(command);
+
+		cancellationToken.ThrowIfCancellationRequested();
+		await OnExecutingAsync(command);
+		
+		var cultureCodes = command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.DistinctBy(d=>d.CultureCode).Select(d=>CultureCode.From(d.CultureCode)).ToList();
+		
+		var localizedEntities = await DbContext.{{contextProperty}}Localized.Where(x => cultureCodes.Contains(x.CultureCode)).AsNoTracking().ToListAsync(cancellationToken);
+		
+		var entities = new List<{{enumEntity}}>();
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=> !localizedEntities.Any(e=>e.Id == Enumeration.FromDatabase(dto.Id) && e.CultureCode == CultureCode.From(dto.CultureCode))).ForEach(dto =>
+		{
+			var e = new {{enumEntity}} {Id = Enumeration.FromDatabase(dto.Id), CultureCode = CultureCode.From(dto.CultureCode), Name = dto.Name};
+			DbContext.Entry(e).State = EntityState.Added;
+			entities.Add(e);
+		});
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=> localizedEntities.Any(e=>e.Id == Enumeration.FromDatabase(dto.Id) && e.CultureCode == CultureCode.From(dto.CultureCode))).ForEach(dto =>
+		{
+			var e = new {{enumEntity}} {Id = Enumeration.FromDatabase(dto.Id), CultureCode = CultureCode.From(dto.CultureCode), Name = dto.Name};
+			DbContext.Entry(e).State = EntityState.Modified;
+			entities.Add(e);
+		});
+		
+		command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s.Where(dto=>dto.CultureCode == _defaultCultureCode.Value).ForEach(dto =>
+		{
+			var e = new {{enumAtt.EntityNameForEnumeration}} { Id = Enumeration.FromDatabase(dto.Id), Name = dto.Name };
+			DbContext.Entry(e).State = EntityState.Modified;
+		});
+		
+
+		await OnBatchCompletedAsync(command, localizedEntities);
+		await DbContext.SaveChangesAsync(cancellationToken);
+		return command.{{enumAtt.EntityDtoNameForLocalizedEnumeration}}s;
```

Comment:
We don't need to return the Request ... we need to provide the situation in the Database after this Upsert operation.